backend/src/controllers/auth-controller.ts

```diff
@@ -51,8 +79,21 @@ export const loginUser = async (req: Request, res: Response) => {
         const password = req.body.password as string;
         User.findOne({email: email})
             .then(user => {
-                compare(password, user!.passwordHash/*forgot how to fix possibly null*/).then(result => {
-                    res.status(200).json({ message: 'Success', isPasswordCorrect: result });
+                if (!user) {
+                    res.status(404).json({ message: "User does not exist." });
+                    return
+                }
+                compare(password, user.passwordHash/*forgot how to fix possibly null*/).then(result => {
```

Comment:
That's okay. We can assume `passwordHash` is not null because `userSchema` requires `passwordHash` to exist and the API only creates users with a populated `passwordHash` field

Reply:
const result = await compare(...)

...